userdocs/diagnostics/protocol-type-non-conformance.md

```diff
@@ -1,46 +1,59 @@
-# Protocol type not conforming to itself
-Protocols in Swift may be used as types. Protocols as types are sometimes called existential types.
+# Protocol Types Cannot Conform to Protocols
 
+In Swift, a protocol that does not have `Self` or associated type requirements can be used as a type. You can use a variable or constant of a protocol type, also called an __existential type__, to hold a value of any conforming type:
```

Comment:
I think it should be “contravariant Self”, because this works:

```swift
protocol P {
  func foo() -> Self
}

struct S: P { 
  func foo() -> S {
    self
  }
}

let p: P = S() // okay
```

Reply:
The details are more applicable to the other educational note on PATs. However, I'm a bit surprised that this works, since `var foo: Self { get }` does not.

Reply:
I'm not sure if I'm thinking on the right track, just a thought to throw about:

One of the differences between functions and computed variables is that only functions can throw errors. `Error` is the only protocol in Swift that has self-conformance. Is there any special handling added to functions because of `Error`, that might explain the discrepancy?

Reply:
It's not just `Error`.  `Any` and many Obj-C protocols are also self-conformant.